src/Repository/VoucherRepository.php

```diff
@@ -41,6 +41,29 @@ public function countUnredeemedVouchers(): int
         return $this->matching(Criteria::create()->where(Criteria::expr()->eq('redeemedTime', null)))->count();
     }
 
+
+    /**
+     * Returns the number of redeemed vouchers per user.
+     */
+    public function countRedeemedVouchersByUser(User $user): int
```

Comment:
Can you put both queries in one method and give them a optional second parameter `$redeemed = false`?